### ISSUE TITLE
some terms appearing multiple times in scratch file on `update`

### ISSUE THREAD

**Issue by aryairani** [unisonweb/unison#4436]
from https://unisonlanguage.slack.com/archives/CLUNF0J5S/p1701108041098149
> I'm hitting an issue where when I make some changes in my scratch file and run update it prints some of the terms in my codebase to the scratch file multiple times, so I get errors that there are multiple definitions. As far as I can tell the implementations are the same, so it doesn't seem to be because of a merge conflict (also this is happening on update, not merge). I assume that this should never happen?

**Comment by aryairani** [comment]
on hold pending repro

**Comment by kylegoetz** [comment]
Any possibility you've done some updates to transitive dependencies and you have intermediate dependencies that haven't fully migrated to the new transitive ones? So if you were to `view` the code that's givin gyou problems, and then view the code that is directly used, eventually you find some type or term that isn't named but instead is just a hash?

I had that issue, where the code was I thought un-changed in my scratch, but one was calling a function with a term of type

`baz = _projects.098qw4euoasdgjhaqosegoasdgoashdg.foo.bar -> ...` while the other was calling `baz = foo.bar -> ...`. The previous one indicated the dependency hadn't been updated to properly call new `baz` but was trying to call the old `baz`, which was a different type because the underlying `foo.bar` type had been updated, but something went wrong with the migration (maybe there was a pending `todo` to handle that was caused by `update.old`?

I had a *lot* of these, but I'm almost positive they all arose because I was updating things but never ran `todo` until a few weeks later when I realized that was a part of the `update` process, and by then so many things had changed that they'd, I suppose, escaped the algorithm's calculation of what is left to do.
